File: AOJ/0285/0285.c
#include<stdio.h>

int j, y;
char path[64];

void parse(int a, int b){
  if ( a == j && b == y ){
    path[a + b] = '\0';
    printf("%s\n", path); return;
  } else if ( a == 5 && b <= 3 || b == 5 && a <= 3 ){
    return;
  }
  if ( a > j || b > y ) return;
  path[a + b] = 'A';
  parse(a + 1, b);
  path[a + b] = 'B';
  parse(a, b + 1);
}

int main(){
  scanf("%d %d", &j, &y);
  parse(0, 0);
  return 0;
}

File: AOJ/0285/0285.py
j,y = map(int, input().split())

def parse(a, b, path):
    if a == j and b == y:
        print(path)
        return
    elif a == 5 and b <= 3 or b == 5 and a <= 3:
        return
    if  a > j or b > y:
        return
    parse(a + 1, b, path + "A")
    parse(a, b + 1, path + "B")

parse(0, 0, "")
